recipe_estimator.py

```diff
@@ -41,6 +41,31 @@ def add_ingredients_to_solver(ingredients, solver, total_ingredients):
 
     return ingredient_numvars
 
+# Add constraints to ensure that the quantity of each ingredient is greater than or equal to the quantity of the next ingredient
+# and that the sum of children ingredients is equal to the parent ingredient
+def add_relative_constraints_on_ingredients(solver, parent_ingredient_numvar, ingredient_numvars):
```

Comment:
I'm not sure that this is a good way to do it. It might be better just to keep ingredients without a CIQAL code completely out of the optimisation and then just add them back in at the end, half way between their greater and lesser ingredient

Reply:
Well we do need the solver to know about ingredients without CIQUAL code even if we consider that they don't have any nutrient contribution, as they add constraints on the other ingredients.
e.g. if we have "unknown, sugar, oil", then sugar must be 50% or less.